pygmt/helpers/testing.py

```diff
@@ -152,8 +151,13 @@ def load_static_earth_relief():
     data : xarray.DataArray
         A grid of Earth relief for internal tests.
     """
-    fname = which("@static_earth_relief.nc", download="c")
-    return load_dataarray(fname)
+    with Session() as lib:
+        with lib.virtualfile_out(kind="grid") as voutgrd:
+            lib.call_module(
+                module="read", args=["@static_earth_relief.nc", voutgrd, "-Tg"]
+            )
+            grid = lib.virtualfile_to_raster(vfname=voutgrd, kind="grid")
```

Comment:
You know what? Maybe we could think of refactoring `pygmt.io.load_dataarray` to use this with block? It would be a breaking change, but that could mean we can use GMT's I/O directly.

Reply:
Yes, actually, I'm thinking if we should provide a wrapper for the special `read` module, which can read a table, grid, or image, and return pd.DataFrame or xarray.DataArray. The function looks like:
```
def read(file, kind=None, **kwargs):
    if kind == None:
        # Try our best to decide the data kind. Or make 'kind' a required parameter
        pass 

    code = {"dataset": "d", "grid": "g", "image": "i"}[kind]
    func = 

    with Session() as lib:
        with lib.virtualfile_out(kind=kind) as voutfile:
            lib.call_module(
                module="read", args=[file, voutfile, f"-T{code}"]
            )
            if kind == "dataset":
                return lib.virtualfile_to_dataset(vfname=voutfile, **kwargs)
            elif kind in {"grid", "image"}
                return lib.virtualfile_to_raster(vfname=voutgrd, kind=kind)
```
For comparison, GMT.jl provides the [`gmtread`](https://www.generic-mapping-tools.org/GMTjl_doc/documentation/utilities/gmtread/index.html)/[`gmtwrite`](https://www.generic-mapping-tools.org/GMTjl_doc/documentation/utilities/gmtwrite/index.html#gmtwrite) functions, but As I understand it, the functions return the GMT_DATASET/GMT_GRID/GMT_IMAGE containers, rather than data structures in the Julia world.